messaging/src/main/java/org/axonframework/common/infra/DescribableComponent.java

```diff
@@ -18,7 +18,22 @@
 
 import jakarta.annotation.Nonnull;
 
+/**
+ * Contract allowing components {@link #describeTo(ComponentDescriptor) to describe} themselves with a given
+ * {@link ComponentDescriptor}.
+ *
+ * @author Allard Buijze
+ * @author Mitchel Herrijgers
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+@FunctionalInterface
 public interface DescribableComponent {
 
+    /**
+     * Describe the properties of {@code this DescribableComponent} with the given {@code descriptor}.
```

Comment:
Reading your [question](https://github.com/AxonFramework/AxonFramework/pull/3338#discussion_r2005566015) on #3338, I think it would be good if the `describeTo` operation does say what the good practices are when implementing this method.

Reply:
Added, please re-check that.

Reply:
Great write up man, thanks!